### PR TITLE
Improved documentation on simple_functions.factorial()

### PR DESCRIPTION
**Pull request by ojhall94**
Added some checks on factorial():
- Input value has a type hint
- Raises valueerror if input is not an integer
- Raises valueerror if input is not positive
- Added docstring

### CODE REVIEW

File: Hall/template/simple_functions.py [hunk 1]
```diff
@@ -5,7 +5,28 @@ def fibonacci(max):
     return values
```
Comment (sjschmidt) [inline Hall/template/simple_functions.py:7]: Reviewing!

Comment (sjschmidt) [reply Hall/template/simple_functions.py:7]: Looks good to me.

**Review by sjschmidt — APPROVED** [review]

**Comment by ojhall94** [comment]
Thanks!

**Comment by bsipocz** [comment]
@ojhall94 @sjschmidt - Thanks for opening the pull request and for the review(s)! Everything looks good, so we'll go ahead and merge this. Note that we are not conducting a review of the code itself, because the exercise was about the overall process not the details of the code. 

@sjschmidt - your comments are generic and not about the specific code changes in the "Files changed" tab. For these cases it's possible to leave the generic comments when you set the review status (Approve/Require Changes/etc.), or in the conversation tab, here rather than inline.

**Pull request merged**